parlai/utils/misc.py

```diff
@@ -349,17 +351,81 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         pass
 
 
-def nice_report(report):
+def _report_sort_key(report_key: str) -> Tuple[str, str]:
+    """
+    Sorting name for reports.
+
+    Sorts by main metric alphabetically, then by task.
+    """
+    fields = report_key.split("/")
+    main_key = fields.pop(-1)
```

Comment:
Main key is the metric name, e.g fields can be like "empathetic_dialogues" and main_key will be "f1". the reason it could be multiple levels is sometimes names have /'s in them (like filenames)